### ISSUE TITLE
Getting 'signed out' a lot : "You are not allowed to be here"

### ISSUE THREAD

**Issue by alex-wilmer** [orionjs/core#19]
What is triggering this message? Is there something I can do to prevent this error from popping up so frequently? It will confuse the admin's for sure.

**Comment by nicolaslopezj** [comment]
I'm having this error too. I will try to fix it

**Comment by nicolaslopezj** [comment]
I really need help here.. I not sure what is causing this. If you know what detonates de logout please tell.

**Comment by pcorey** [comment]
This is only happening when I open an entity for updating (and maybe creation/deletion), but it happens fairly frequently.

Something is republishing Meteor.users without some fields required by your hasPermission helper (specifically, permission and isAdmin). To see what I mean, add this on the client:

    Tracker.autorun(function() {
        console.log('users', Meteor.users.find().fetch());
    });

And throw a log statement into your hasPermission Meteor.users helper:

    Meteor.users.helpers({
        hasPermission: function(key, strict) {
            console.log('this.isAdmin', this.isAdmin);

As you go about using Orion, you'll notice the this.isAdmin log spitting out trues. Eventually, you'll open an entity, your Meteor.users computation block will rerun and you'll notice that your user object is missing a lot of fields. this.isAdmin will be undefined, and hasPermission will fail.

Still looking into what's causing this...

**Comment by pcorey** [comment]
It looks like the way to pass additional fields into Meteor.user() and Meteor.users is to do a null publication. Adding these to server.js seems to fix the issue:

    Meteor.publish(null, function() {
        if (this.userId) {
            return Meteor.users.find(
                {_id: this.userId}, {
                    fields: {
                        profile: 1,
                        username: 1,
                        emails: 1,
                        permission: 1,
                        isAdmin: 1
                    }
                }
            );
        } else {
            return null;
        }
    });

    Accounts.addAutopublishFields({
        forLoggedInUser: ['permission', 'isAdmin'],
        forOtherUsers: []
    });

I can do a pull request, but I'm not going to pretend I fully understand this fix...

Check out [accounts-base](https://github.com/chmac/meteor-accounts-base/blob/master/accounts_server.js#L1010-L1029) for more info. And check out what some of the accounts providers do when [autopublish is on](https://github.com/meteor/meteor/blob/devel/packages/accounts-facebook/facebook.js#L15-L25).

**Comment by nicolaslopezj** [comment]
Wow, thanks!

**Comment by pcorey** [comment]
I figured out what was causing this, if anyone's interested:

isAdmin and permission were being added to the Meteor.users collection by the adminUsers publication. Eventually, by switching back and forth between routes enough, the number of subscriptions in SubManager exceeded the cache limit, and adminUsers was removed. This caused the Meteor.users collection to be populated by the default unnamed publication in accounts-base (which only published standard user fields).

Unlike named publications, unnamed publications can be defined multiple times for a single collection. The union of the fields published by each of these unnamed publications is what ends up being published to the client.

So, by creating our own unnamed publication and publishing __at least__ permission and isAdmin, we can be sure that these fields will be passed down to Meteor.users along with whatever fields accounts-base published to that collection.

Whew...
